packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1692,3 +1764,15 @@ Hardhat Network's forking functionality only works with blocks from at least spu
     });
   }
 }
+
+function getCommon(networkId: number, blockNumber: number): Common | undefined {
+  try {
+    const common = Common.forCustomChain(networkId, {});
+
+    common.setHardfork(common.activeHardfork(blockNumber));
+
+    return common;
+  } catch (e) {
+    return undefined;
```

Comment:
IIRC we decided to throw an error in this case. As this feature will only work for the "standard" chains, the ones included in `@ethereumjs/common`.

Reply:
Fixed on b326b89ee34484362a1e03ed34ce553fbbdadedb. I also renamed `getCommon` to `getCommonForTracing`. I'm not sure about the error message.